src/shims/mod.rs

```diff
@@ -1,3 +1,5 @@
+#![warn(clippy::integer_arithmetic)]
```

Comment:
Not sure how useful this one is... none of he cases are caught, so we could just as well turn on debug asserts in release mode if we haven't already. Would be more readable than the checked math functions

Reply:
> turn on debug asserts in release mode

Any suggestion for how to do that in a way that also applies to the builds produced by rustc? We cannot use cargo profiles.

Also we do have a bunch of arithmetic outside this folder (stacked borrows, data races, weak memory) that I deliberately did not enable this lint for, so there might be a performance difference. But that should be easy to benchmark.

Reply:
We can change Bootstrap's build flags for miri

Reply:
Yea I don't expect this to make a difference. It's mostly in math heavy code where this is visible in practice

Reply:
> We can change Bootstrap's build flags for miri

How? RUSTFLAGS affects all dependencies so it'll rebuild everything...

I am going to move the contentious part into a separate PR.